parcels/collection/collectionnodes.py

```diff
@@ -0,0 +1,1782 @@
+        assert len(self._data) > 0
+        result = ParticleCollectionNodes(self._idgen, self._c_lib_register, self._pclass, lon=np.empty(shape=0), lat=np.empty(shape=0), depth=np.empty(shape=0), time=np.empty(shape=0), pid_orig=None, lonlatdepth_dtype=self._lonlatdepth_dtype, ngrid=self._ngrid)
+        for id in ids:
+            ndata = self.pop_single_by_ID(id)
+            pdata = ndata.data
+            ndata.set_data(None)
+            del ndata
+            result.add_single(pdata)
+        return result
+
+    def __iadd__(self, same_class):
+        """
+        Performs an incremental addition of the equi-structured ParticleCollections, such to allow
+
+        a += b,
+
+        with 'a' and 'b' begin the two equi-structured objects (or: 'b' being and individual object).
+        This operation is equal to an in-place addition of (an) element(s).
+
+        this is the former "add(same_class)" function.
+        :arg pcollection: second ParticleCollectionNodes object to be merged into this collection
+        :returns vector-list of indices of all merged particles
+        """
+        self.merge_same(same_class)
+        return self
+
+    def insert(self, obj, index=None):
+        """
+        This function allows to 'insert' a Particle (as object or via its accessor) into this collection. This method
+        needs to be specified to each collection individually. Some collections (e.g. unordered list) allow to define
+        the index where the object is to be inserted. Some collections can optionally insert an object at a specific
+        position - at a significant speed- and memory malus cost (e.g. vectors, arrays, dense matrices). Some
+        collections that manage a specified indexing order internally (e.g. ordered lists, sets, trees), and thus
+        have no use for an 'index' parameter. For those collections with an internally-enforced order, the function
+        mapping equates to:
+
+        insert(obj) -> add_single(obj)
+
+        This collection does not evaluate the index, as it is not an indexable collection.
+        :arg obj: Particle object to insert
+        """
+        return self.add_single(obj)
+
+    def push(self, particle_obj):
+        """
+        This function pushes a Particle (as object or via its accessor) to the end of a collection ('end' definition
+        depends on the specific collection itself). For collections with an inherent indexing order (e.g. ordered lists,
+        sets, trees), the function just includes the object at its pre-defined position (i.e. not necessarily at the
+        end). For the collections, the function mapping equates to:
+
+        int32 push(particle_obj) -> add_single(particle_obj); return -1;
+
+        This function further returns the index, at which position the Particle has been inserted. By definition,
+        the index is positive, thus: a return of '-1' indicates push failure, NOT the last position in the collection.
```

Comment:
Hmm, this may be confusing to users. Why this choice? Why not return `None` if the push failed?